### ISSUE TITLE
Require element ID / investigation on 'Custom URLs' button under 'URL Autocomplete Menu'

### ISSUE THREAD

**Issue by npark-mozilla** [mozilla-mobile/focus-android#2555]
### Steps to reproduce
- Execute CustomTabTest.java on API 23

### Expected behavior
- Test should find Custom URLs menu
<img width="330" alt="screen shot 2018-04-28 at 3 01 27 pm" src="https://user-images.githubusercontent.com/6930774/39396786-0ff5b12e-4af5-11e8-82f1-967e9169c7a0.png">

### Actual behavior
- Test cannot find it.

The 'Custom URLs' menu cannot seem to be found by just using the .withText() call in Espresso.
So currently it uses following statement to locate:
```
DataInteraction CustomURLRow = onData(anything())
                .inAdapterView(allOf(withId(android.R.id.list),
                        childAtPosition(
                                withId(android.R.id.list_container),
                                0)))
                .atPosition(4);
CustomURLRow.perform(click());
```
But this fails only in API 23.  In API 23, below works, but then again it will fail on every other android API.
```
        DataInteraction CustomURLRow = onData(anything())
                .inAdapterView(Matchers.allOf(withId(android.R.id.list),
                        childAtPosition(
                                withClassName(is("android.widget.LinearLayout")),
                                0)))
                .atPosition(4);
        CustomURLRow.perform(click());
```
we need an identifier for the element in order to make things easier & run tests on all applicable api.

@ekager if you have time, could you take a look at this please?  I'll work on fixing the test when the ID is added.

**Comment by ekager** [comment]
Looking into this on Monday and assigning myself to remember

**Comment by boek** [comment]
So I took a look at this and it looks like the Preferences API doesn't allow you to assign an ID to anything. @pocmo Is there a preferred way to fix this? Or should I change it to a button?

**Comment by pocmo** [comment]
@npark-mozilla Yeah, I remember we had to close a similar issue in the past. Do you remember how we solved this? Unfortunately we do not have much control over how Android builds the settings UI. Do you think there are other options how we can select that item? I also wonder why we do onData(anything()) here. Did we try onData(withText(..)) and does this try to scroll?

**Comment by npark-mozilla** [comment]
Hmm I don't remember how we fixed it - I think we worked it around at the time by using the index of the elements.  I'm seeing this kind of issues on 'Block * trackers' switch in new Privacy & Security submenu as well. 
` UiObject blockAnalyticTrackerEntry = TestHelper.settingsList.getChild(new UiSelector()
                .className("android.widget.LinearLayout")
                .instance(2));
`
and
`
        UiObject blockSocialTrackerEntry = TestHelper.settingsList.getChild(new UiSelector()
                .className("android.widget.LinearLayout")
                .instance(3));
`
refers to same element (analytic tracker switch), although uiautomatorviewer says the latter one points to the social tracker block switch. I have to change the .instance() index to  4 from 3 to toggle the social tracker.  Really a myster.
